src/Microsoft.AspNetCore.Authentication/AuthenticationHandler.cs

```diff
@@ -142,7 +142,7 @@ protected string BuildRedirectUri(string targetPath)
                     Logger.AuthenticationSchemeNotAuthenticated(Scheme.Name);
                 }
             }
-            else if (result?.Failure != null)
+            else
```

Comment:
result can never be null?

Reply:
if it is null then first "if" will be true (line 133).

Reply:
Ah!